src/table/cells/handle-key-press.js

```diff
@@ -67,48 +57,71 @@ export const preventDefaultBehavior = (evt) => {
   evt.preventDefault();
 };
 
-export const moveFocus = (evt, rootElement, rowIndex, colIndex, selState) => {
+export const moveFocus = (evt, rootElement, cellCoord, setFocusedCell, selState, page) => {
   preventDefaultBehavior(evt);
   removeCurrentFocus(evt);
   const rowAndColumnCount = getRowAndColumnCount(rootElement);
-  const { nextRow, nextCol } = arrowKeysNavigation(evt, rowAndColumnCount, rowIndex, colIndex, selState);
-  moveToNextFocus(rowAndColumnCount.rowElements, nextRow, nextCol);
+  const nextCellCoord = arrowKeysNavigation(evt, rowAndColumnCount, cellCoord, selState);
+  focusCell(rowAndColumnCount.rowElements, nextCellCoord);
+  nextCellCoord.push(page);
+  setFocusedCell(nextCellCoord);
 };
 
-export const headHandleKeyPress = (evt, rootElement, rowIndex, colIndex, changeSortOrder, layout, isDim) => {
+export const headHandleKeyPress = (
+  evt,
+  rootElement,
+  cellCoord,
+  setFocusedCell,
+  changeSortOrder,
+  layout,
+  isDim,
+  page
+) => {
   switch (evt.key) {
     case 'ArrowUp':
     case 'ArrowDown':
     case 'ArrowRight':
     case 'ArrowLeft': {
-      !isCtrlShift(evt) && moveFocus(evt, rootElement, rowIndex, colIndex);
+      !isCtrlShift(evt) && moveFocus(evt, rootElement, cellCoord, setFocusedCell, page);
       break;
     }
     // Space bar / Enter: update the sorting
     case ' ':
     case 'Enter': {
       preventDefaultBehavior(evt);
-      changeSortOrder(layout, isDim, colIndex);
+      changeSortOrder(layout, isDim, cellCoord[1]);
       break;
     }
     default:
       break;
   }
 };
 
-export const bodyHandleKeyPress = (evt, rootElement, rowIndex, colIndex, cell, selState, selDispatch) => {
+export const bodyHandleKeyPress = (
+  evt,
+  rootElement,
+  cellCoord,
+  focusedCell,
+  setFocusedCell,
+  cell,
+  selState,
+  selDispatch,
+  page
+) => {
   switch (evt.key) {
     case 'ArrowUp':
     case 'ArrowDown':
     case 'ArrowRight':
     case 'ArrowLeft': {
-      !isCtrlShift(evt) && moveFocus(evt, rootElement, rowIndex, colIndex, selState);
+      !isCtrlShift(evt) && moveFocus(evt, rootElement, cellCoord, setFocusedCell, selState, page);
       break;
     }
     // Space bar: Selects value.
     case ' ': {
       preventDefaultBehavior(evt);
-      cell?.isDim && selectCell(cell, selState, selDispatch, evt);
+      cell?.isDim &&
+        handleCellFocus(cell, focusedCell, setFocusedCell, rootElement, page) &&
```

Comment:
fixed the case before clicking,  first tabing into the chart and move a couple of cells

Reply:
why do you need `page`?

Reply:
`focusedCell[2] === page && removeTabSequence(rootElement.getElementsByClassName('sn-table-row'), focusedCell)`

to check the page that a user is selecting by clicking or space key is the same as the page of the focused cell or not.
If it is the same, set tabIndex to -1. Otherwise, no need to do that.

Reply:
But it is still the same cell, on main at least. If I navigate to a cell with the keyboard, then go to the next page, that last focus cell still has tabindex 0. In any case, passing around `page` this way is not worth it. Setting tabindex -1 one to many times shouldn't be that performance heavy.

Reply:
You are right. And I came up with a way to address the issue when removing `page` by `cell && cell.setAttribute('tabIndex', '-1');`

Reply:
When we change the` rows per page` and sometimes `page`, we will lose focus. it should be created in a new issue.